src/bosh-softlayer-cpi/softlayer/common/vm_utils.go

```diff
@@ -43,28 +43,34 @@ func CreateVirtualGuestTemplate(stemcell bslcstem.Stemcell, cloudProps VMCloudPr
 		switch network.Type {
 		case "dynamic":
 			if value, ok := network.CloudProperties["PrimaryNetworkComponent"]; ok {
+				networkVlan := sldatatypes.NetworkVlan{}
 				networkComponent := value.(map[string]interface{})
 				if value1, ok := networkComponent["NetworkVlan"]; ok {
-					networkValn := value1.(map[string]interface{})
-					if value2, ok := networkValn["Id"]; ok {
-						cloudProps.PrimaryNetworkComponent = sldatatypes.PrimaryNetworkComponent{
-							NetworkVlan: sldatatypes.NetworkVlan{
-								Id: int(value2.(float64)),
-							},
-						}
+					networkVlanInfo := value1.(map[string]interface{})
+					if value2, ok := networkVlanInfo["Id"]; ok {
```

Comment:
Do you mind separating these into two statements? I find these `if x, y: = statement(); ok { ... }` more difficult to parse and less clear than just having more statement. This is especially true since you have many nested ones.

Perhaps also having some small private functions to do the various checks and set better?